pkg/image/decompress.go

```diff
@@ -40,15 +41,23 @@ func UnpackData(filename string, src io.ReadCloser) (io.ReadCloser, error) {
 // return a ReadCloser.  It is up to the caller of DecompressData to close the returned stream.
 // If no compression is detected, it is considered a 'noop' and the original stream is returned.
 // Returns trimmed filename string and gzip Reader if gzip compression was used.
-func DecompressData(filename string, src io.ReadCloser) (string, io.ReadCloser, error) {
+func DecompressData(filename string, src io.ReadCloser) (fn string, rc io.ReadCloser, err error) {
 	glog.Infof("DecompressData: checking if %q is compressed\n", filename)
 	ext := filepath.Ext(TrimString(filename))
 	switch ext {
-	case ExtGz:
-		glog.Infof("DecompressData: detected %v compression format", ExtGz)
-		filename = strings.TrimSuffix(filename, ext) // trim ".gz"
-		src, err := gunzip(src)
-		return filename, src, err
+	case ExtGz, ExtXz:
```

Comment:
The outer switch seems unnecessary. Why not 
```
switch ext {    
case ExtGZ:
        rc, err = gunzip(src)
case ExtXz:
	rc, err = xzDecompress(src)
default:
        rc = src
}
if err != nil {
    return "", nil, fmt.Errorf("DecompressData: %v\n", err)
}
fn = strings.TrimSuffix(filename, ext)
return fn, rc, nil
```

Reply:
The outer switch is there to allow a single call to `glog.Infof()`.